### DISCUSSION TITLE
[Feat]: Autoplay - Add pause autoplay method

### DISCUSSION THREAD

**Discussion by jasongerbes** [davidjerleke/embla-carousel#1021] [Ideas] [answered]
### Which variants of Embla Carousel are relevant to this feature request?

- [ ] embla-carousel **(Core)**
- [ ] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [X] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-fade
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

### Feature description

I'd like to implement a play/pause control similar to this:

<img width="307" alt="image" src="https://github.com/user-attachments/assets/25aa658f-909a-4793-a080-5603cd286ad4">

Autoplay currently has methods for `play()` and `stop()`. However `stop()` resets the autoplay progress rather than pausing it.

Would you be open to adding a `pause()` method?

### Additional Context

Additional details here...

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues
- [X] I agree to follow this project's Contributing Guidelines for feature requests

**Comment by pras-wjy** [comment]
i have a hacky way for this, if you want to pause you can just remove the plugin.

this is how i implement it in react:
```tsx
import React, { useState } from 'react'
import useEmblaCarousel from 'embla-carousel-react'
import AutoPlay from "embla-carousel-autoplay"

export function EmblaCarousel() {
  const [isAutoPlay, setIsAutoPlay] = useState(true)
  const [emblaRef, emblaApi] = useEmblaCarousel(
    {loop: false},
    [
      isAutoPlay ? AutoPlay({ delay: 2000 }) : ...[] // spread empty list so it returns nothing
    ] 
  )

  return (
    <div>
      <div className="embla" ref={emblaRef}>
        <div className="embla__container">
          <div className="embla__slide">Slide 1</div>
          <div className="embla__slide">Slide 2</div>
          <div className="embla__slide">Slide 3</div>
        </div>
      </div>
      <button onClick={() => seIsAutoPlay(!isAutoPlay)}>
        {isAutoPlay ? "Pause" : "Play"}
      </button>
    </div>
  )
}
```

**Comment by davidjerleke** [comment]
- @jasongerbes @prs-wjy thanks for the feature request. I've added this to the roadmap #841.

**Comment by danielolivier** [comment]
Any updates on this?

**Reply by davidjerleke** [reply]
@danielolivier not yet.

**Reply by davidjerleke** [reply, marked as answer]
@jasongerbes, @danielolivier Implemented in https://github.com/davidjerleke/embla-carousel/commit/bb262c42f1040882aed1af6247431bc92a99c23a. To be released with v9.

**Reply by m-b-1g** [reply]
Hi! when is v9 getting released?

**Reply by davidjerleke** [reply]
Hi @m-b-1g,

I don’t have an exact release date for **v9** yet, but I’m actively working on it — you can follow the progress here:
🔗 [Commit activity](https://github.com/davidjerleke/embla-carousel/commits/feature/v9/)

That said, I’m working on it in the time I have available. If you’d like to help speed things up, you’re very welcome to sponsor the project:
💖 [GitHub Sponsors](https://github.com/sponsors/davidjerleke)

Thanks for your interest and support!

**Reply by davidjerleke** [reply]
@danielolivier @jasongerbes @pras-wjy @jasongerbes,

`v9.0.0-rc01` has now been released which includes the **pause method**:

- [Link to release](https://github.com/davidjerleke/embla-carousel/releases/tag/v9.0.0-rc01)
- [Pause docs](https://www.embla-carousel.com/plugins/autoplay/#pause)
